src/runtime/recipe/recipe-resolver.ts

```diff
@@ -20,25 +20,33 @@ import {Recipe, IsValidOptions} from './recipe.js';
 import {ConnectionConstraint, InstanceEndPoint} from './connection-constraint.js';
 import {SlotConnection} from './slot-connection.js';
 import {SlotUtils} from './slot-utils.js';
+import {Continuation} from './walker.js';
 
 export class ResolveWalker extends RecipeWalker {
+  options: IsValidOptions;
   private readonly arc: Arc;
 
-  constructor(tactic, arc) {
+  constructor(tactic, arc, options?) {
     super(tactic);
     this.arc = arc;
+    this.options = options;
   }
 
-  onHandle(recipe: Recipe, handle: Handle) {
+  onHandle(recipe: Recipe, handle: Handle): Continuation<Recipe, Handle[]> {
+    const error = (label: string) => {
+      if (this.options && this.options.errors) {
+        this.options.errors.set(handle, label);
+      }
+      return []; // Avoiding repetition here allows us to change the type.
```

Comment:
I don't understand the comment. How do you avoid the repetition, and what type can be changed? The return type of this lambda? In what way?

Reply:
This was a note to myself during refactor work on this code. It had previously returned undefined (which was not documented in the type) and using this lambda to simplify the code also let me change to using an empty list instead of the undefined. As it is no longer useful I'll remove it.